lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -303,7 +303,51 @@ getSemanticResults(SILFunctionType *functionType, IndexSubset *parameterIndices,
 static CanSILFunctionType
 getAutoDiffDifferentialType(SILFunctionType *originalFnTy,
                             IndexSubset *parameterIndices, unsigned resultIndex,
-                            LookupConformanceFn lookupConformance) {
+                            LookupConformanceFn lookupConformance,
+                            TypeConverter &TC) {
+  auto getTangentParameterConvention =
+      [&](CanType tanType,
+          ParameterConvention origParamConv) -> ParameterConvention {
+    tanType =
+        tanType->getCanonicalType(originalFnTy->getSubstGenericSignature());
+    AbstractionPattern pattern(originalFnTy->getSubstGenericSignature(),
+                               tanType);
+    auto &tl =
+        TC.getTypeLowering(pattern, tanType, TypeExpansionContext::minimal());
+    if (tl.isAddressOnly() && !isIndirectFormalParameter(origParamConv)) {
```

Comment:
An explanatory comment on top of this `if` would be nice to help readers: something about handling address-only `TangentVector` types. It took me a minute to think through why this `if` is necessary.

Same regarding the `if` below in `getTangentResultConvention`.

Reply:
Done